lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
+
+      // Match category of URN
+      switch (cat) {
+        case 'pbk:ec:secp256r1':
+          // Check for compressed/uncompressed ECDSA public keys
+          if (charArr[0] === '02' || charArr[0] === '03') {
+            // Check length of compressed public key
+            if (charArr.length !== 25) {
+              return false;
+            }
+          } else if (charArr[0] === '04') {
+            // Check length of uncompressed public key
+            if (charArr.length !== 65) {
+              return false;
+            }
+          }
+
+          break;
+        case 'pbk:rsa:2048':
+          // Grab RSA exponent
+          var exponent = id.substring(id.length - 6);
+
+          // Check if RSA exponent is in binary format or if the total ID length is proper size (512 bytes + 6 bits)
+          if (!exponent.matches("[01]+") || id.length !== 518) {
+            return false;
+          }
+
+          break;
+
+        // BLE ids are standardized with a 6 bytes MAC address
+        case 'ble:1\.0':
+          if (charArr.length !== 6) {
+            return false;
```

Comment:
It's a good practice to provide reason of rejection. Maybe throwing an error is the way to go.